src/SFML/Audio/AudioDevice.cpp

```diff
@@ -484,6 +572,60 @@ std::optional<ma_device_id> AudioDevice::getSelectedDeviceId() const
 ////////////////////////////////////////////////////////////
 bool AudioDevice::initialize()
 {
+    // Create the context
+    m_context.emplace();
+
+    auto contextConfig                         = ma_context_config_init();
+    contextConfig.pLog                         = &*m_log;
+    std::uint32_t                  deviceCount = 0;
+    const auto                     nullBackend = ma_backend_null;
+    std::vector<const ma_backend*> backendLists;
+
+    if (!getCurrentDeviceSelection().useNull)
+        backendLists.push_back(nullptr);
+
+    backendLists.push_back(&nullBackend);
+
+    for (const auto* backendList : backendLists)
+    {
+        // We can set backendCount to 1 since it is ignored when backends is set to nullptr
+        if (const auto result = ma_context_init(backendList, 1, &contextConfig, &*m_context); result != MA_SUCCESS)
+        {
+            m_context.reset();
+            err() << "Failed to initialize the audio playback context: " << ma_result_description(result) << std::endl;
+            return false;
+        }
+
+        // Count the playback devices
+        if (const auto result = ma_context_get_devices(&*m_context, nullptr, &deviceCount, nullptr, nullptr);
+            result != MA_SUCCESS)
+        {
+            err() << "Failed to get audio playback devices: " << ma_result_description(result) << std::endl;
+            return false;
+        }
+
+        // Check if there are audio playback devices available on the system
+        if (deviceCount > 0)
+            break;
+
+        // Warn if no devices were found using the default backend list
+        if (backendList == nullptr)
+            err() << "No audio playback devices available on the system" << std::endl;
+
+        // Clean up the context if we didn't find any devices
+        ma_context_uninit(&*m_context);
+    }
+
+    // If the NULL audio backend also doesn't provide a device we give up
+    if (deviceCount == 0)
+    {
+        m_context.reset();
+        return false;
+    }
+
+    if (!getCurrentDeviceSelection().useNull && (m_context->backend == ma_backend_null))
+        err() << "Using NULL audio backend for playback" << std::endl;
```

Comment:
As selecting the NULL device is a state the user can choose, the error message is no longer sufficient, as it doesn't indicate that the NULL device is used despite a different selection.

Additionally, one might consider using the term "device" instead of "audio backend"